common/resource/fx.go

```diff
@@ -215,6 +216,20 @@ func NamespaceRegistryProvider(
 	)
 }
 
+func PersistenceHealthSignalAggregatorProvider(
```

Comment:
I would add a feature flag for next release to control if the HealthSignalAggregator is enabled or not (use no-op impl) in case there's performance implication in production. After we are confident with the performance, we can remove that feature flag.